.github/workflows/ci.yml

```diff
@@ -0,0 +1,194 @@
+      if: |
+        steps.cache-global-package-manager-cache.outputs.cache-hit != 'true' ||
+        steps.cache-node-modules.outputs.cache-hit != 'true'
+    - name: Test
+      run: yarn test:ember --launch ${{ matrix.browser }}
+
+  floating-dependencies:
+    name: Floating Dependencies
+    runs-on: ${{ matrix.os }}
+    needs: lint
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        browser: [chrome]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - uses: actions/setup-node@v2-beta
+      with:
+        node-version: '10.x'
+    - name: Get package manager's global cache path
+      id: global-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - name: Cache Yarn cache
+      id: cache-global-package-manager-cache
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.global-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-yarn-
+    - name: Cache node_modules
+      id: cache-node-modules
+      uses: actions/cache@v1
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-
+    - name: Install Dependencies
+      run: yarn install --frozen-lockfile
+      if: |
+        steps.cache-global-package-manager-cache.outputs.cache-hit != 'true' ||
+        steps.cache-node-modules.outputs.cache-hit != 'true'
+    - name: Test
+      run: yarn test:ember --launch ${{ matrix.browser }}
+
+  try-scenarios:
+    name: Tests - ${{ matrix.ember-try-scenario }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.allow-failure }}
+    needs: test
+
+    strategy:
+      fail-fast: true
+      matrix:
+        ember-try-scenario: [
+          ember-lts-3.4,
+          ember-lts-3.8,
+          ember-lts-3.12,
+          ember-lts-3.16,
+          ember-lts-3.20,
+          ember-release,
+          ember-beta,
+          ember-default-with-jquery,
+          ember-classic
+        ]
+        allow-failure: [false]
+        include:
+          - ember-try-scenario: ember-canary
+            allow-failure: true
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - uses: actions/setup-node@v2-beta
+      with:
+        node-version: '10.x'
+    - name: Get package manager's global cache path
+      id: global-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - name: Cache Yarn cache
+      id: cache-global-package-manager-cache
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.global-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}
```

Comment:
For the workflow job involving `ember-try`, I think we need to have a unique cache for each scenario.

https://gist.github.com/ijlee2/25004d700cd008ee08ecc4873c109a5b#file-ci-yml-L224-L225

Reply:
Can you elaborate why? I thought `ember-try` expects all dependencies installed as defined in `package.json` and `yarn.lock` / `package-lock.json` when started. It will then modify `package.json` as defined for the scenario and run `yarn install` as part of it's own execution. At the end it will cleanup again. Could this nevertheless pollute the cache?